emission/ce186/utility_model.py

```diff
@@ -0,0 +1,505 @@
+
+    def update_to_now(self):
+        self.update_times(datetime.datetime.now())
+
+    def normalize_sounds(self):
+        counter = emmc.Counter()
+        for k,v in self.time_to_noise.iteritems():
+            counter[k] = v
+        counter.normalize()
+        self.time_to_noise = counter
+
+
+def in_bounding_box(lat, lon, bounding_box):
+    return bounding_box[1][0] <= lat and lat <= bounding_box[0][0] and bounding_box[0][1] <= lon and lon <= bounding_box[1][1]
+    
+def parse_noise():
+    noise_file = open("emission/ce186/noise_data.csv")
+    sproul_noises, glade_noises, wellmen_noises, leconte_noises = {}, {}, {}, {}
+    sproul_tl, sproul_br = (37.870637,-122.259722), (37.868926,-122.259005)
+    glade_tl, glade_br = (37.87359,-122.260098), (37.872707,-122.258687)
+    wellmen_tl, wellmen_br = (37.873045,-122.263377), (37.872501,-122.261803)
+    lecont_tl, lecont_br = (37.873278,-122.256959), (37.872277,-122.25639)
+    time = datetime.datetime(2040, 10, 10, 6, 0, 0)
+    td = datetime.timedelta(minutes=10)
+    for l in noise_file:
+        l = l.split(',')
+        sproul_noises[time] = float(l[0])
+        glade_noises[time] = float(l[1])
+        wellmen_noises[time] = float(l[2])
+        leconte_noises[time] = float(l[3])
+    sproul = Area("sproul", sproul_tl, sproul_br, time_to_noise=sproul_noises)
+    glade = Area("glade", glade_tl, glade_br, time_to_noise=glade_noises)
+    wellmen = Area("wellmen", wellmen_tl, wellmen_br, time_to_noise=wellmen_noises)
+    leconte = Area("leconte", lecont_tl, lecont_br, time_to_noise=leconte_noises)
+    return [sproul, glade, wellmen, leconte]
+    
+
+def parse_beauty():
+    beauty_file = open("emission/ce186/beauty.csv")
+    beauty_areas = [ ]
+    for beauty_line in beauty_file:
+        beauty_line = beauty_line.split(',')
+        name = beauty_line[0]
+        tl = (float(beauty_line[1]), float(beauty_line[2]))
+        br = (float(beauty_line[5]), float(beauty_line[6]))
+        beauty = int(beauty_line[9])
+        a = Area(name, tl, br, beauty=beauty)
+        beauty_areas.append(a)
+    return beauty_areas
+
+def get_noise_score(lat, lng, noises, time):
+    tot = 0
+    to_return = 0
+    for noise_area in noises:
+        if noise_area.point_in_area(lat, lng):
+            to_return += get_closest(time, area)
+    if to_return > 0:
+        return to_return
+    return .5 ## if point isnt in any mapped area return the average
+
+def get_closest(time, area):
+    for k, v in area.time_to_noise.iteritems():
+        if time - k < datetime.timedelta(minutes=10):
+            return v
+
+def normalize_scores(areas):
+    counter = emmc.Counter()
+    #print "len of areas is %s" % len(areas)
+    for area in areas:
+        if area.beauty:
+            counter[area.name] = area.beauty
+        elif area.noise:
+            counter[area.name] = area.noise
+    counter.normalize()
+    
+    new_areas = [ ]
+    for name, value in counter.iteritems():
```

Comment:
I don't understand what you are doing here. Do you have some comments? if you have a set of areas already why do you need to iterate through them and create new areas?